rules/sha/index.bzl

```diff
@@ -0,0 +1,10 @@
+# Calculates a sha of the contents of the files in `srcs` and outputs it to a file called `name`.txt.
+def sha(name, srcs, **kwargs):
+    native.genrule(
+        name = name,
+        srcs = srcs,
+        outs = [name + ".txt"],
+        cmd = "echo \"$$(echo \"$(SRCS)\" | xargs find | sort | xargs shasum | shasum | awk '{ print $$1 }')\" > $@",
```

Comment:
nit: I think you could remove the subshell and run

```
echo \"$(SRCS)\" | xargs find | sort | xargs shasum | shasum | awk '{ print $$1 }' > $@
```

Reply:
Also, uber nit: run `xargs` with `--delimiter '\n'`

Reply:
> nit: I think you could remove the subshell
Done


> Also, uber nit: `xargs --delimiter '\n' find` in case file names contain spaces
SRCS is space delimited. Looks like this is a more general bazel problem: https://github.com/bazelbuild/bazel/issues/4327